extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMusicSearchExtractor.java

```diff
@@ -0,0 +1,527 @@
+                break;
+            case MUSIC_ARTISTS:
+                params = "Eg-KAQwIABAAGAAgASgAMABqChAEEAUQAxAKEAk%3D";
+                break;
+            default:
+                params = null;
+                break;
+        }
+
+        // @formatter:off
+        byte[] json = JsonWriter.string()
+            .object()
+                .object("context")
+                    .object("client")
+                        .value("clientName", "WEB_REMIX")
+                        .value("clientVersion", youtubeMusicKeys[2])
+                        .value("hl", "en")
+                        .value("gl", getExtractorContentCountry().getCountryCode())
+                        .array("experimentIds").end()
+                        .value("experimentsToken", "")
+                        .value("utcOffsetMinutes", 0)
+                        .object("locationInfo").end()
+                        .object("musicAppInfo").end()
+                    .end()
+                    .object("capabilities").end()
+                    .object("request")
+                        .array("internalExperimentFlags").end()
+                        .object("sessionIndex").end()
+                    .end()
+                    .object("activePlayers").end()
+                    .object("user")
+                        .value("enableSafetyMode", false)
+                    .end()
+                .end()
+                .value("query", getSearchString())
+                .value("params", params)
+            .end().done().getBytes("UTF-8");
+        // @formatter:on
+
+        final Map<String, List<String>> headers = new HashMap<>();
+        headers.put("X-YouTube-Client-Name", Collections.singletonList(youtubeMusicKeys[1]));
+        headers.put("X-YouTube-Client-Version", Collections.singletonList(youtubeMusicKeys[2]));
+        headers.put("Origin", Collections.singletonList("https://music.youtube.com"));
+        headers.put("Referer", Collections.singletonList("music.youtube.com"));
+        headers.put("Content-Type", Collections.singletonList("application/json"));
+
+        final Response response = getDownloader().post(url, headers, json);
+
+        if (response.responseCode() == 404) {
+            throw new ContentNotAvailableException("Not found" +
+                    " (\"" + response.responseCode() + " " + response.responseMessage() + "\")");
+        }
+
+        final String responseBody = response.responseBody();
+        if (responseBody.length() < 50) { // ensure to have a valid response
+            throw new ParsingException("JSON response is too short");
+        }
+
+        final String responseContentType = response.getHeader("Content-Type");
+        if (responseContentType != null && responseContentType.toLowerCase().contains("text/html")) {
+            throw new ParsingException("Got HTML document, expected JSON response" +
+                    " (latest url was: \"" + response.latestUrl() + "\")");
+        }
```

Comment:
The only thing now are these duplicated checks from `YoutubeParsingHelper`, maybe we could extract them to yet another general response check in there?

Reply:
Yeah, I already thought it wasn't that clean.